bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/SomfyTahomaHandlerFactory.java

```diff
@@ -157,16 +159,14 @@ protected ThingHandler createHandler(Thing thing) {
     protected synchronized void removeHandler(ThingHandler thingHandler) {
         if (thingHandler instanceof SomfyTahomaBridgeHandler) {
             ServiceRegistration<?> serviceReg = this.discoveryServiceRegs.get(thingHandler.getThing().getUID());
-            if (serviceReg != null) {
-                // remove discovery service, if bridge handler is removed
-                SomfyTahomaItemDiscoveryService service = (SomfyTahomaItemDiscoveryService) bundleContext
-                        .getService(serviceReg.getReference());
-                if (service != null) {
-                    service.deactivate();
-                }
-                serviceReg.unregister();
-                discoveryServiceRegs.remove(thingHandler.getThing().getUID());
+            // remove discovery service, if bridge handler is removed
+            SomfyTahomaItemDiscoveryService service = (SomfyTahomaItemDiscoveryService) bundleContext
+                    .getService(serviceReg.getReference());
```

Comment:
This is the old way of having a thing bound service.

If you have a minute, could you adapt this to the new way? (A thing handler nowadays has a `getServices()` method. Overwrite it, return `SomfyTahomaItemDiscoveryService.class`, implement the suggested interface. Thanks.